packages/backend/src/util/decorateObjectWithResources.ts

```diff
@@ -0,0 +1,67 @@
+import type { CreateBackendApiOptions, Organization, Session, User } from '../api';
+import { createBackendApiClient } from '../api';
+import type { AuthObject } from '../tokens/authObjects';
+
+type DecorateAuthWithResourcesOptions = {
+  loadSession?: boolean;
+  loadUser?: boolean;
+  loadOrganization?: boolean;
+};
+
+type WithResources<T> = T & {
+  session?: Session;
+  user?: User;
+  organization?: Organization;
+};
+
+export const decorateObjectWithResources = async <T extends object>(
+  obj: T,
+  authObj: AuthObject,
+  apiClientOptions: CreateBackendApiOptions,
+  opts: DecorateAuthWithResourcesOptions,
+): Promise<WithResources<T>> => {
+  const { loadSession, loadUser, loadOrganization } = opts || {};
+  const { userId, sessionId, orgId } = authObj;
+
+  const { sessions, users, organizations } = createBackendApiClient({ ...apiClientOptions });
+
+  const [sessionResp, userResp, organizationResp] = await Promise.all([
+    loadSession && sessionId ? sessions.getSession(sessionId) : Promise.resolve(undefined),
+    loadUser && userId ? users.getUser(userId) : Promise.resolve(undefined),
+    loadOrganization && orgId ? organizations.getOrganization({ organizationId: orgId }) : Promise.resolve(undefined),
+  ]);
+
+  const session = sessionResp && !sessionResp.errors ? sessionResp.data : undefined;
+  const user = userResp && !userResp.errors ? userResp.data : undefined;
+  const organization = organizationResp && !organizationResp.errors ? organizationResp.data : undefined;
+
+  return Object.assign(obj, sanitizeAuthObject({ session, user, organization }));
+};
+
+/**
+ * @internal
+ */
+export function sanitizeAuthObject<T extends Record<any, any>>(authObject: T): T {
```

Comment:
❓ Should we rename this to something more specific such as `safeAuthObject` or `stripPrivateMatadata(authObject: T)`?

BTW we can be more specific in the types here and use the three AuthObject types instead of `Record<any, any>`

Reply:
done!